lib/test/KettleTestUtils.http.js

```diff
@@ -51,9 +54,86 @@ fluid.defaults("kettle.test.request", {
     termMap: {}
 });
 
+// A component that sends multipart/form-data via POST
+// using the form-data module
+// Can be used to send both files and fields
+// Used currently for testing the multer-based middleware
+//
+// TODO: Should elements of this and kettle.test.request.http be merged?
```

Comment:
Yes, please derive this from kettle.test.request.http, and make the following refactoring of kettle.test.request.http's antediluvian workflow:
i) convert the "send" invoker into an event named "send", which accepts an options object rather than the broken-out five arguments
i) a) Leave the existing invoker as a backwards-compatible driver that just forwards to a firing of the event
ii) take the body of the existing "send" invoker and split it into two namespaced listeners
a) prepareRequest containing everything up to and including binding req.on("error") excluding the payload, and give this a priority of "before:sendPayload"
b) sendPayload including the preparation of the payload and sending it to the request

Then your formData grade just consists of an overriding of the sendPayload listener, together with a use of the "alternative submission method" listed here https://github.com/form-data/form-data#alternative-submission-methods in order to reuse the already created "nativeRequest"